### ISSUE TITLE
2.0 and 2.1 don't work, but the 1.0 and 1.1 Appimage work properly

### ISSUE THREAD

**Issue by zipperzbieracz** [PaulCombal/SamRewritten-legacy#130]
I'm new to this and totally green, but the 2.0 and 2.1 Appimage versions don't work for me, but the 1.0 and 1.1 versions work properly. When I try to launch the 2.1 version in the terminal to see what is happening, this pops up:
___________
(samrewritten:477943): Gtk-WARNING **: 17:55:23.389: Could not load a pixbuf from icon theme.
This may indicate that pixbuf loaders or the mime database could not be found.
**
Gtk:ERROR:../../../../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /usr/share/icons/Mint-Y/actions/16/image-missing.svg: Nie można wczytać modułu wczytującego obrazy: /usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-svg.so: /lib/x86_64-linux-gnu/librsvg-2.so.2: undefined symbol: cairo_font_options_get_variations (gdk-pixbuf-error-quark, 5)
Bail out! Gtk:ERROR:../../../../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /usr/share/icons/Mint-Y/actions/16/image-missing.svg: Nie można wczytać modułu wczytującego obrazy: /usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-svg.so: /lib/x86_64-linux-gnu/librsvg-2.so.2: undefined symbol: cairo_font_options_get_variations (gdk-pixbuf-error-quark, 5)
/tmp/.mount_SamRewfmApgu/AppRun: linia 8: 477943 Przerwane               (zrzut pamięci) ${SCRIPT_PATH}/samrewritten $@
___________
Nie można wczytać modułu wczytującego obrazy = can't load modules loading pictures
Przerwane = interrupted
zrzut pamięci = memory dump

**Comment by PaulCombal** [comment]
Does it work if you compile it yourself?

**Comment by zipperzbieracz** [comment]
Yes I managed to compile it myself and the most recent version works with all functionalities.

**Comment by PaulCombal** [comment]
Glad this works! I can see where this comes from.. Unfortunately I'm just waiting for the next version of the graphics library to be available to do a big update, so I won't fix this for now and can only wait. Feel free to close the issue if necessary 👍

**Comment by mikeydislog** [comment]
> Glad this works! I can see where this comes from.. Unfortunately I'm just waiting for the next version of the graphics library to be available to do a big update, so I won't fix this for now and can only wait. Feel free to close the issue if necessary +1

Sorry to high jack this thread but is there any way to get this working on the Deck?  I currently have no other means of computer for a few months and for whatever reason, Steam in itself over written with what I had on my Deck and lost achievements on a specific game and it's just annoying me that they are locked again, and a game that isn't exactly easy either...not a big deal of course, as its just a game, but still annoying.   I don't know how to compile etc (i did try reading but it's very alien to me) so i used the appimage but getting that symbol error similar to what was mentioned here.

**Comment by PineApple1242** [comment]
3 years later, me on Pop!_OS, the glib hasn't been updated, hence I cannot open the app image despite running it as an executable. 

How would I go on about compiling it myself? If there is a guide I can get linked, I'd appreciate it!

**Comment by PaulCombal** [comment]
There's no specific instructions for Pop_OS but you can find the general guidelines on the Readme: https://github.com/PaulCombal/SamRewritten?tab=readme-ov-file#building
